### ISSUE TITLE
Surface overtagging on separated foot and bike paths

### ISSUE THREAD

**Issue by apophenic-treehugger** [streetcomplete/StreetComplete#3222]
SC encourages users to overtag the surface and segregated bike and foot paths by asking them for cycleway:surface and footway:surface even if they are identical (most common case). This even happens if surface=* is already set.
See discussion in https://www.openstreetmap.org/changeset/109767128

**How to Reproduce**
I'm not a SC user but an OSM mapper. This behaviour has been reported by a SC user who's not me (see above OSM changeset).

**Versions affected**
current

**Comment by matkoniecz** [comment]
See https://github.com/streetcomplete/StreetComplete/pull/1493#discussion_r307422894

**Comment by westnordost** [comment]
Well, it's not a bug and it is not leading to that any data is destroyed. It only leads to some duplicate information.

From the changeset discussion:

> Otherwise, this only leads to overtagging with no gain in information; which unnecessarily makes features harder to understand and to maintain.

The gain in information is that specifically the surfaces of the segregated foot- and cycleway have been checked to be indeed the same. I don't understand the argument that it is harder to maintain.

**Comment by apophenic-treehugger** [comment]
Ok fair enough, I see your point.

**Harder to maintain** because this _duplicate information_ may need to be changed at some point. Imagine this case, which I think is more common than separate surfaces:
Before: `surface=gravel`
Then reconstruction
After: `surface=asphalt`
This would be easier to maintain (also without accidentally leaving outdated tags) if there is only one `surface=*` tag instead of 3 (identical ones) to be changed.

But ok, I think we all agree it doesn't cause astronomical amounts of harm, so I'm fine with leaving it as it if the effort is not worth changing it.

I'm glad this kind of user friendly encouraging app exists.

**Comment by westnordost** [comment]
Thanks!

Hmm well, I can't imagine an unpaved (foot)way tagged with `segregated=yes`. StreetComplete only asks about the cycleway/footway surface on segregated ways, so this particular example doesn't work.

**Comment by apophenic-treehugger** [comment]
Ok, point given. (I realized that after posting)
To give a more realistic example, I think `segregated=yes` + `surface=paving_stones` -> `surface=asphalt` is a common evolution.

But as already said, it's really a minor issue and I the last thing I want is discourage you guys from continuing your great work by grumbling around.

**Comment by Helium314** [comment]
It still might be possible to reduce the number of unnecessary tags by not asking for surface if a way is elegible for the segregated quest. Or at least give the segregated quest higher default priority than path surface.

**Comment by westnordost** [comment]
> It still might be possible to reduce the number of unnecessary tags by not asking for surface if a way is elegible for the segregated quest

Right now, the surface quest (tagging `surface=*`) is already not asked for ways that are segregated. So, StreetComplete either asks "what is the surface of this way" **or** "what is the surface of the cycleway-part of this way".

StreetComplete also asks whether a given way is segregated or not. If your suggestion is to move that quest up so that it is asked before asking for any surface by default: Hmm, good idea! Will do that.

**Comment by westnordost** [comment]
Alright, so I will conclude this report with this small enhancement, thanks everyone

**Comment by westnordost** [comment]
StreetComplete prefers **_explicit_** tags if something has been checked **_explicitly_**. For example, it prefers setting `cycleway:left/right` over `cycleway`, even if the cycleway situation on both sides is the same. (Though in this case, it uses `cycleway:both`)

**Comment by apophenic-treehugger** [comment]
Sorry, I don't see how this relates to my comment. Could you explain?

**Comment by apophenic-treehugger** [comment]
Oh, ok, on second though, I do see.

**Comment by Helium314** [comment]
> > It still might be possible to reduce the number of unnecessary tags by not asking for surface if a way is elegible for the segregated quest
> 
> Right now, the surface quest (tagging `surface=*`) is already not asked for ways that are segregated. So, StreetComplete either asks "what is the surface of this way" **or** "what is the surface of the cycleway-part of this way".
> 
> StreetComplete also asks whether a given way is segregated or not. If your suggestion is to move that quest up so that it is asked before asking for any surface by default: Hmm, good idea! Will do that.

I just noticed that segregation is only asked for paved surfaces, so moving the _segregated_ quest before surface does not change anything...

**Comment by westnordost** [comment]
Hm, right. I'll revert that change then to avoid confusion.

I think it is better to have rather too many `surface` tags on paths than too many `segregated` tags (on unpaved paths), so I think current implementation is good.

**Comment by smichel17** [comment]
> this happens quite often (5%? 10%?).

Perhaps you could get some real data to see how often it is. Could you write an overpass query ways which have been "overtagged" by SC? Then you could see how many confirmed the existing tagging vs added new info.

**Comment by apophenic-treehugger** [comment]
That's a great idea, @smichel17 !

If it turns out there is substantial wrong tagging, I don't think it would be the correct approach to add additional, contradicting/confusing tags. We might end up with cases like `surface:paved` + `cycleway:surface=asphalt` + `footway:surface=gravel`, which seems ambiguous. Data consumer will be left wondering what the `surface:paved` tag refers to.

Instead, if we don't trust preexisting `surface=*` tags (e. g. because a survey as proposed by michel shows that we shouldn't), then how about using SC's verification functionality the same way users are asked to verify opening hours, etc? If the app finds `highway=path` + `bicycle=designated` + `foot=designated` + `segregated=yes` + `surface=paving_stones`, then it could ask _"Is it still the case that the surface of **both** the cycleway **and** the footway are still paving stones?"_
In case of a positive answer, a tag like `surface:check_date=*` could be added (not sure whether this is the right key to use) to make the check date explicit.
If the answer is _no_, then remove the `surface=*` tag (and optionally continue to ask which surface both the footway and cycleway have).

**Comment by westnordost** [comment]
Could we use a new issue for that? This one is getting kinda long and has been closed already.

Am 1. September 2021 10:10:43 MESZ schrieb apophenic-treehugger ***@***.***>:
>That's a great idea, @smichel17 !
>
>If it turns out there is substantial wrong tagging, I don't think it
>would be the correct approach to add additional,
>contradicting/confusing tags. We might end up with cases like
>`surface:paved` + `cycleway:surface=asphalt` +
>`footway:surface=gravel`, which seems ambiguous. Data consumer will be
>left wondering what the `surface:paved` tag refers to.
>
>Instead, if we don't trust preexisting `surface=*` tags (e. g. because
>a survey as proposed by michel shows that we shouldn't), then how about
>using SC's verification functionality the same way users are asked to
>verify opening hours, etc? If the app finds `highway=path` +
>`bicycle=designated` + `foot=designated` + `segregated=yes` +
>`surface=paving_stones`, then it could ask _"Is it still the case that
>the surface of **both** the cycleway **and** the footway are still
>paving stones?"_
>In case of a positive answer, a tag like `surface:check_date=*` could
>be added (not sure whether this is the right key to use) to make the
>check date explicit.
>If the answer is _no_, then remove the `surface=*` tag (and optionally
>continue to ask which surface both the footway and cycleway have).
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/streetcomplete/StreetComplete/issues/3222#issuecomment-910047118

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by apophenic-treehugger** [comment]
Absolutely. #3238
